lite2/client.go

```diff
@@ -734,16 +734,17 @@ func (c *Client) sequence(
 			if err != nil {
 				return errors.Wrapf(err, "failed to obtain the vals #%d", height+1)
 			}
-		}
-		err = c.updateTrustedHeaderAndNextVals(interimHeader, interimNextVals)
-		if err != nil {
-			return errors.Wrapf(err, "failed to update trusted state #%d", height)
+			if !bytes.Equal(trustedHeader.NextValidatorsHash, interimNextVals.Hash()) {
```

Comment:
```suggestion
			if !bytes.Equal(interimHeader.NextValidatorsHash, interimNextVals.Hash()) {
```

?

Reply:
we should have a test for it!

Reply:
true the test only picked up the bisection method (skipping)